libs/sr-db/src/Database/SroDataManager.ts

```diff
@@ -0,0 +1,61 @@
+import { DataManagerBase } from '@genshin-optimizer/database'
+import type { ISrObjectDescription } from '@genshin-optimizer/sr-srod'
+import type { ISroDatabase, SroDatabase } from '..'
+import type { ImportResult } from './exim'
+export class SroDataManager<
+  CacheKey extends string,
+  DataKey extends string,
+  CacheValue extends StorageValue,
+  StorageValue
+> extends DataManagerBase<
+  CacheKey,
+  DataKey,
+  CacheValue,
+  StorageValue,
+  SroDatabase
+> {
+  constructor(database: SroDatabase, dataKey: DataKey) {
+    super(database, dataKey)
+    // If the storage has a key for some entry AND
+    // the entry doesn't exist in memory:
+    // Delete it from storage
+    for (const key of this.database.storage.keys)
+      if (
+        key.startsWith(this.goKeySingle) &&
+        !this.set(this.toDatumKey(key), {})
+      ) {
+        this.database.storage.remove(key)
+      }
+  }
+  exportSROD(sro: Partial<ISrObjectDescription & ISroDatabase>) {
+    const key = this.dataKey.replace('sro_', '')
+    sro[key] = Object.entries(this.data).map(([id, value]) => ({
+      ...this.deCache(value),
+      id,
+    }))
+  }
+  importSROD(sro: ISrObjectDescription & ISroDatabase, _result: ImportResult) {
+    const entries = sro[this.dataKey]
+    if (entries && Array.isArray(entries))
+      entries.forEach((ele) => ele.id && this.set(ele.id, ele))
+  }
+  override get goKeySingle() {
+    const key = this.dataKey.replace('sro_', '')
+    if (key.endsWith('s')) return key.slice(0, -1)
+    return key
+  }
+  override toStorageKey(key: string): string {
+    return `${this.goKeySingle}${key}`
+  }
+  override toDatumKey(key: string): CacheKey {
+    return key.split(this.goKeySingle)[1] as CacheKey
+  }
+  override generateKey(keys: Set<string> = new Set(this.keys)): string {
```

Comment:
This override means that relic IDs will be `50` instead of `relic_50`. Same goes for light cones. I don't think this will be an issue, but something to note

Reply:
Alternatively, we change this function generate with toStorageKey, and override toDatumKey in relic and light cone managers to just return key rather than splitting